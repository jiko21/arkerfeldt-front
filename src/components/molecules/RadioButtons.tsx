/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import React from 'react';
import { Color } from '@/const/color';

const wrapperStyle = css`
  input {
    display: none;
  }

  input + label {
    color: ${Color.LIGHT_TEXT};
    padding-left: 20px;
    position: relative;
    margin-right: 20px;
  }

  input + label:before {
    content: '';
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    width: 12px;
    height: 12px;
    border: 1px solid ${Color.BLACK};
    border-radius: 50%;
  }

  input:checked + label {
    color: ${Color.ENABLE};
    font-weight: 700;
  }

  input:checked + label:after {
    content: '';
    display: block;
    position: absolute;
    top: 4px;
    left: 4px;
    width: 6px;
    height: 6px;
    background: ${Color.ENABLE};
    border-radius: 50%;
  }
`;

const labelStyle = css`
  margin-left: 12px;
`;

interface IProps {
  items: string[];
  value: string;
  name: string;
  onChangeHandler: (value: string) => void;
}

const RadioButtons: React.FC<IProps> = ({
  items,
  value,
  name,
  onChangeHandler,
}) => (
  <>
    {items.map((item) => (
      <div key={item} css={wrapperStyle}>
        <input
          type="radio"
          id={item}
          name={name}
          value={item}
          onChange={() => {
            onChangeHandler(item);
          }}
          checked={item === value}
          data-testid={`radio-btn-${item}`}
        />
        <label css={labelStyle} htmlFor={item}>
          {item}
        </label>
      </div>
    ))}
  </>
);

export default RadioButtons;
