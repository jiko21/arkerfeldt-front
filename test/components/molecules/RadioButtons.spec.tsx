import RadioButtons from '@/components/molecules/RadioButtons';
import { fireEvent, render, screen } from '@testing-library/react';

describe('CheckBox.tsx', () => {
  let onChangeHandlerMock: jest.Mock;
  beforeEach(() => {
    onChangeHandlerMock = jest.fn();
  });

  afterEach(() => {
    onChangeHandlerMock.mockClear();
  });
  it('should correctly render (checked: true)', () => {
    const props = {
      value: 'すべて',
      name: 'status',
      onChangeHandler: onChangeHandlerMock,
      items: ['すべて', '公開中', '未公開'],
    };
    const { container } = render(<RadioButtons {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should correctly call onClick', () => {
    const props = {
      value: 'すべて',
      name: 'status',
      onChangeHandler: onChangeHandlerMock,
      items: ['すべて', '公開中', '未公開'],
    };
    render(<RadioButtons {...props} />);
    fireEvent.click(screen.getByTestId('radio-btn-公開中'));
    expect(onChangeHandlerMock).toBeCalledWith('公開中');
  });
});
