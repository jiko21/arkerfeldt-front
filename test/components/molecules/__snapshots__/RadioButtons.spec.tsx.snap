// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox.tsx should correctly render (checked: true) 1`] = `
.emotion-0 input {
  display: none;
}

.emotion-0 input+label {
  color: #6A6A6A;
  padding-left: 20px;
  position: relative;
  margin-right: 20px;
}

.emotion-0 input+label:before {
  content: '';
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 12px;
  height: 12px;
  border: 1px solid #2d2d2d;
  border-radius: 50%;
}

.emotion-0 input:checked+label {
  color: #007AAE;
  font-weight: 700;
}

.emotion-0 input:checked+label:after {
  content: '';
  display: block;
  position: absolute;
  top: 4px;
  left: 4px;
  width: 6px;
  height: 6px;
  background: #007AAE;
  border-radius: 50%;
}

.emotion-1 {
  margin-left: 12px;
}

<div>
  <div
    class="emotion-0"
  >
    <input
      checked=""
      data-testid="radio-btn-すべて"
      id="すべて"
      name="status"
      type="radio"
      value="すべて"
    />
    <label
      class="emotion-1"
      for="すべて"
    >
      すべて
    </label>
  </div>
  <div
    class="emotion-0"
  >
    <input
      data-testid="radio-btn-公開中"
      id="公開中"
      name="status"
      type="radio"
      value="公開中"
    />
    <label
      class="emotion-1"
      for="公開中"
    >
      公開中
    </label>
  </div>
  <div
    class="emotion-0"
  >
    <input
      data-testid="radio-btn-未公開"
      id="未公開"
      name="status"
      type="radio"
      value="未公開"
    />
    <label
      class="emotion-1"
      for="未公開"
    >
      未公開
    </label>
  </div>
</div>
`;
